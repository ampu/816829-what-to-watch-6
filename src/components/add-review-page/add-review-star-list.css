.rating__input:disabled + .rating__label {
  cursor: default;
}

.rating__input:disabled + .rating__label::after {
  opacity: 0.5;
}
