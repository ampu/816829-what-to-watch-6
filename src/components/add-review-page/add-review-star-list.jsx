import React, {memo} from 'react';
import PropTypes from 'prop-types';

const AddReviewStarList = ({maxRating}) => {
  return Array.from({length: maxRating}, (_, i) => {
    const value = i + 1;
    const id = `star-${value}`;

    return (
      <React.Fragment key={id}>
        <input className="rating__input" id={id} type="radio" name="rating" value={value}/>
        <label className="rating__label" htmlFor={id}>Rating {value}</label>
      </React.Fragment>
    );
  });
};

AddReviewStarList.propTypes = {
  maxRating: PropTypes.number.isRequired,
};

export {AddReviewStarList};

export default memo(AddReviewStarList);
