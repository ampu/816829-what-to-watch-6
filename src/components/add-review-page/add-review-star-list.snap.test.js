import React from 'react';

import {render} from '@testing-library/react';

import {AddReviewStarList} from './add-review-star-list';

describe(`AddReviewStarList component`, () => {
  it(`Should render according to the snapshot`, () => {
    const {container} = render(<AddReviewStarList maxRating={10}/>);
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot when disabled`, () => {
    const {container} = render(<AddReviewStarList maxRating={10} isDisabled/>);
    expect(container).toMatchSnapshot();
  });
});
