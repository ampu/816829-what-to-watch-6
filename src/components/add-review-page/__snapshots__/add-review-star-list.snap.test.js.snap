// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReviewStarList component Should render according to the snapshot 1`] = `
<div>
  <input
    class="rating__input"
    id="star-1"
    name="rating"
    type="radio"
    value="1"
  />
  <label
    class="rating__label"
    for="star-1"
  >
    Rating 
    1
  </label>
  <input
    class="rating__input"
    id="star-2"
    name="rating"
    type="radio"
    value="2"
  />
  <label
    class="rating__label"
    for="star-2"
  >
    Rating 
    2
  </label>
  <input
    class="rating__input"
    id="star-3"
    name="rating"
    type="radio"
    value="3"
  />
  <label
    class="rating__label"
    for="star-3"
  >
    Rating 
    3
  </label>
  <input
    class="rating__input"
    id="star-4"
    name="rating"
    type="radio"
    value="4"
  />
  <label
    class="rating__label"
    for="star-4"
  >
    Rating 
    4
  </label>
  <input
    class="rating__input"
    id="star-5"
    name="rating"
    type="radio"
    value="5"
  />
  <label
    class="rating__label"
    for="star-5"
  >
    Rating 
    5
  </label>
  <input
    class="rating__input"
    id="star-6"
    name="rating"
    type="radio"
    value="6"
  />
  <label
    class="rating__label"
    for="star-6"
  >
    Rating 
    6
  </label>
  <input
    class="rating__input"
    id="star-7"
    name="rating"
    type="radio"
    value="7"
  />
  <label
    class="rating__label"
    for="star-7"
  >
    Rating 
    7
  </label>
  <input
    class="rating__input"
    id="star-8"
    name="rating"
    type="radio"
    value="8"
  />
  <label
    class="rating__label"
    for="star-8"
  >
    Rating 
    8
  </label>
  <input
    class="rating__input"
    id="star-9"
    name="rating"
    type="radio"
    value="9"
  />
  <label
    class="rating__label"
    for="star-9"
  >
    Rating 
    9
  </label>
  <input
    class="rating__input"
    id="star-10"
    name="rating"
    type="radio"
    value="10"
  />
  <label
    class="rating__label"
    for="star-10"
  >
    Rating 
    10
  </label>
</div>
`;

exports[`AddReviewStarList component Should render according to the snapshot when disabled 1`] = `
<div>
  <input
    class="rating__input"
    disabled=""
    id="star-1"
    name="rating"
    type="radio"
    value="1"
  />
  <label
    class="rating__label"
    for="star-1"
  >
    Rating 
    1
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-2"
    name="rating"
    type="radio"
    value="2"
  />
  <label
    class="rating__label"
    for="star-2"
  >
    Rating 
    2
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-3"
    name="rating"
    type="radio"
    value="3"
  />
  <label
    class="rating__label"
    for="star-3"
  >
    Rating 
    3
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-4"
    name="rating"
    type="radio"
    value="4"
  />
  <label
    class="rating__label"
    for="star-4"
  >
    Rating 
    4
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-5"
    name="rating"
    type="radio"
    value="5"
  />
  <label
    class="rating__label"
    for="star-5"
  >
    Rating 
    5
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-6"
    name="rating"
    type="radio"
    value="6"
  />
  <label
    class="rating__label"
    for="star-6"
  >
    Rating 
    6
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-7"
    name="rating"
    type="radio"
    value="7"
  />
  <label
    class="rating__label"
    for="star-7"
  >
    Rating 
    7
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-8"
    name="rating"
    type="radio"
    value="8"
  />
  <label
    class="rating__label"
    for="star-8"
  >
    Rating 
    8
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-9"
    name="rating"
    type="radio"
    value="9"
  />
  <label
    class="rating__label"
    for="star-9"
  >
    Rating 
    9
  </label>
  <input
    class="rating__input"
    disabled=""
    id="star-10"
    name="rating"
    type="radio"
    value="10"
  />
  <label
    class="rating__label"
    for="star-10"
  >
    Rating 
    10
  </label>
</div>
`;
